src/apps/explorer/pages/AppData/FormPage.tsx

```diff
@@ -0,0 +1,267 @@
+  const [invalidFormDataAttempted, setInvalidFormDataAttempted] = useState<{ appData: boolean; ipfs: boolean }>({
+    appData: false,
+    ipfs: false,
+  })
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [ipfsHashInfo, setIpfsHashInfo] = useState<IpfsHashInfo | void | undefined>()
+  const [ipfsCredentials, setIpfsCredentials] = useState<{ pinataApiKey?: string; pinataApiSecret?: string }>({})
+  const [isDocUploaded, setIsDocUploaded] = useState<boolean>(false)
+  const [error, setError] = useState<string>()
+
+  const network = useNetworkId()
+  const formRef = React.useRef<Form<FormProps>>(null)
+  const ipfsFormRef = React.useRef<Form<FormProps>>(null)
+
+  useEffect(() => {
+    const fetchSchema = async (): Promise<void> => {
+      const latestSchema = await getSchema()
+      setSchema(latestSchema)
+      setAppDataForm(INITIAL_FORM_VALUES)
+    }
+
+    fetchSchema()
+  }, [])
+
+  const toggleInvalid = (data: { [key: string]: boolean }): void => {
+    setInvalidFormDataAttempted((prevState) => ({ ...prevState, ...data }))
+  }
+
+  const resetFormFields = (form: string): void => {
+    setDisabledIPFS(true)
+    setIpfsCredentials({})
+    toggleInvalid({ ipfs: false })
+    if (form === 'appData') {
+      setDisabledAppData(true)
+    }
+  }
+
+  const handleFormatData = (formData: any): any => {
+    if (!formData.metadata || !Object.keys(formData.metadata).length) return formData
+    const formattedData = structuredClone(formData)
+    const isReferrerEnabled = formattedData.metadata.referrer.enableReferrer
+    const isQuoteEnabled = formattedData.metadata.quote.enableQuote
+
+    deletePropertyPath(formattedData, 'metadata.referrer.enableReferrer')
+    deletePropertyPath(formattedData, 'metadata.quote.enableQuote')
+
+    if (!isReferrerEnabled) {
+      deletePropertyPath(formattedData, 'metadata.referrer')
+    }
+    if (!isQuoteEnabled) {
+      deletePropertyPath(formattedData, 'metadata.quote')
+    }
+
+    return formattedData
+  }
+
+  const handleMetadataErrors = (_: FormProps, errors: FormValidation): FormValidation =>
+    handleErrors(formRef, errors, setDisabledAppData)
+
+  const handleIPFSErrors = (_: FormProps, errors: FormValidation): FormValidation =>
+    handleErrors(ipfsFormRef, errors, setDisabledIPFS)
+
+  const handleOnChange = ({ formData }: FormProps): void => {
+    setAppDataForm(formData)
+    if (ipfsHashInfo) {
+      setIpfsHashInfo(undefined)
+      setIsDocUploaded(false)
+      resetFormFields('appData')
+      setError(undefined)
+    }
+    if (JSON.stringify(handleFormatData(formData)) !== JSON.stringify(INITIAL_FORM_VALUES)) {
+      setDisabledAppData(false)
+    }
+  }
+
+  const onSubmit = async ({ formData }: FormProps): Promise<void> => {
+    if (!network) return
+    setIsLoading(true)
+    try {
+      const hashInfo = await COW_SDK[network]?.metadataApi.calculateAppDataHash(handleFormatData(formData))
```

Comment:
The appData calculation actually doesn't depend on the network.
You should be safe using Mainnet until your SDK changes are applied to the Explorer